runtime/runtime_impl_jetty12/src/main/java/com/google/apphosting/runtime/jetty/http/JettyHttpHandler.java

```diff
@@ -123,27 +142,42 @@ public boolean handle(Request request, Response response, Callback callback) thr
       handled = handleException(ex, requestToken, genericResponse);
       Response.writeError(request, response, callback, ex);
     } finally {
-      requestManager.finishRequest(requestToken);
-    }
-    // Do not put this in a final block.  If we propagate an
-    // exception the callback will be invoked automatically.
-    genericResponse.finishWithResponse(context);
-    // We don't want threads used for background requests to go back
-    // in the thread pool, because users may have stashed references
-    // to them or may be expecting them to exit.  Setting the
-    // interrupt bit causes the pool to drop them.
-    if (genericRequest.getRequestType() == RuntimePb.UPRequest.RequestType.BACKGROUND) {
-      Thread.currentThread().interrupt();
+      // We don't want threads used for background requests to go back
+      // in the thread pool, because users may have stashed references
+      // to them or may be expecting them to exit.  Setting the
+      // interrupt bit causes the pool to drop them.
+      if (genericRequest.getRequestType() == RuntimePb.UPRequest.RequestType.BACKGROUND) {
+        Thread.currentThread().interrupt();
```

Comment:
This feels like it is in the wrong spot... 
Can it not be done in the code that dispatches the background thread?

Reply:
I could move it, but then it would be done before the potential error handling in the catch block which could be an issue.